### PR TITLE
feat: Add Temporary Mode

### LINKED ISSUE #10: Support Ephemeral / Temp Mode
One useful use-case for Denali is as an ephemeral catalog for testing purposes. So a temporary database and warehouse. Right now, we can kind of replicate it with the following flags / environment variables:
```shell
DENALI_API_PORT=5000 DENALI_WAREHOUSE_PATH=/tmp/iceberg DENALI_DATABASE_URL=":memory:" DENALI_DATABASE_TYPE="sqlite3" denali start
```

It would be convenient if instead there was a flag that did this automatically, something like (or maybe `--ephemeral`:
```shell
denali start --temp
```

Some additional nice features to include:
- Randomly choose port, retrying if port is occupied
- Randomly generated local warehouse path, cleanup on exit

### PR DESCRIPTION
**Pull request by srilman**
Closes #10. Temporary mode allows users to start up an ephemeral REST catalog for testing purposes with minimal pre-setup. It automatically:

- Chooses a randomly available port (unless an explicit port is provided via the `--port` argument
- Uses an in-memory SQLite server
- Sets up a warehouse in a temporary local path

In addition, this PR contains:
- Options to support random port finding for regular mode
- Some `go.mod` cleanup

### CODE REVIEW

File: logic/state.go [hunk 1]
```diff
@@ -280,6 +280,7 @@ func NewStateFromConfig(config *Config) *State {
 	}
 
 	// Construct an Iceberg IO
+	log.Printf("Using Path `%v` for Iceberg Warehouse", config.Warehouse.Path)
```
Comment (ritwika314) [inline logic/state.go:283]: Is there a reason why we're using `printf` in all these functions as opposed to `println`? Why don't we want these logs in a new line?

File: main.go [hunk 2]
```diff
@@ -30,22 +31,33 @@ var Version = func() string {
...
 	// Create Router and API routes
+	log.Println("Creating Router and API routes")
 	mux := http.NewServeMux()
 	r := routes.Router{Mux: mux, State: s}
 	r.ConfigRoutes()
 	r.NsRoutes()
 	r.ViewRoutes()
 	r.TableRoutes()
 
+	log.Println("Middleware")
```
Comment (ritwika314) [inline main.go]: This log seems a bit terse, is it sort of like a section heading?

File: main.go [hunk 3]
```diff
@@ -91,21 +103,51 @@ Global Flags:
...
 				Flags: []cli.Flag{
 					&cli.Uint64Flag{
-						Name: "port", Aliases: []string{"p"},
+						Name:        "port",
+						Aliases:     []string{"p"},
 						Usage:       "Port to run the REST server on",
-						Value:       0,
-						DefaultText: "config file, or 8080",
+						Value:       0, // 0 means random port
+						DefaultText: "From config file, or randomly chosen if unset",
```
Comment (ritwika314) [inline main.go]: Randomly chosen/generated?

**Review by ritwika314 — APPROVED** [review]
Some minor comments, but LGTM otherwise, thank you!

File: main.go (hunk 3 above)
Comment (srilman) [reply main.go]: Sure, will change to that

File: main.go (hunk 2 above)
Comment (srilman) [reply main.go]: No, thats just me missing a debugging message. Will remove thanks for catching

File: logic/state.go (hunk 1 above)
Comment (srilman) [reply logic/state.go:283]: I believe `log.Printf` implicitly returns a newline at the end. Or at least, when looking at the output, thats what it does. There is no `log.Printfln` or something like that

**Pull request merged**